zipline/pipeline/mixins.py

```diff
@@ -400,14 +400,35 @@ def _compute(self, windows, dates, assets, mask):
 
         real_compute = self._wrapped_term._compute
 
+        if self.windowed:
+            # If we're windowed, inputs are stateful AdjustedArrays.  We don't
+            # need to do any preparation before forwarding to real_compute, but
+            # we need to call `next` on them if we want to skip an iteration.
+            def prepare_inputs():
+                return inputs
+
+            def skip_this_input():
+                for w in inputs:
+                    next(w)
+        else:
+            # If we're not windowed, inputs are just ndarrays.  We need to
+            # slice off one row when forwarding to real_compute, but we don't
+            # need to do anything to skip an input.
+            def prepare_inputs():
+                # i is the loop iteration variable below.
+                return [a[[i]] for a in inputs]
+
+            def skip_this_input():
```

Comment:
I would think that that would encompass the whole action that happens in the loop iteration, whereas this just applies to the part that interacts with `inputs`.